gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java

```diff
@@ -340,6 +340,9 @@
           "fs.gs.inputstream.min.range.request.size",
           GoogleCloudStorageReadOptions.DEFAULT_MIN_RANGE_REQUEST_SIZE);
 
+  public static final HadoopConfigurationProperty<Boolean> SOCKET_KEEP_ALIVE_ENABLE =
+      new HadoopConfigurationProperty<>("fs.gs.socket.keep.alive.enable", true);
+
```

Comment:
Do we need this option? Seems like we may just enable it by default and not complicate configuration. Are there reasons when we do not want to use it?

Reply:
I thnk it is useful when customer runs into complicated issue. 
If their OS TCP timeout is very high, in that case socket won't be closed for long time and this might impact their performance. They can test the connector in those cases.

Reply:
High value for TCP timeouts will have the same effect as not enabling keep alive at all - that's why I think it's fine to hard code this and user can control this behavior on OS-level.

If we do not know about any negative side effects, I think it's better to not add option to customize it to not complicate our code base, we can always add it later if there will be a reason for it.

Reply:
As a point of comparison, it's standard practice for Hadoop RPC and the HDFS DataNode data transfer protocol to enable TCP keep-alive. It's hard-coded, not something that can be toggled off in configuration. I'd be comfortable skipping the configuration property here.

Reply:
Noted. Changed to hard coded true.